### ISSUE TITLE
Interval is always closed

### ISSUE THREAD

**Issue by mbaudin47** [openturns/openturns#2696]
### What is the idea?

The `Interval` class defines a multi-dimensional interval. Its `contains(sample)` method returns a vector boolean which is equal to 1 for each point which is inside the interval and 0 otherwise. One limitation is that the dimension $d$ interval is closed:

$$
\left\[\boldsymbol{a}, \boldsymbol{b}\right\]
$$

for any $\boldsymbol{a}, \boldsymbol{b} \in \mathbb{R}^d$ where $\boldsymbol{a} \leq \boldsymbol{b}$ (componentwise).

In some use-case, I want to define the interval that is closed on the left and open on the right:

$$
\left\[\boldsymbol{a}, \boldsymbol{b}\right).
$$

Indeed, these are the interval which are involved in low discrepancy sequences. Hence, it is required to check that the (t, m, s)-net property is satisfied. Without the feature, the `contains()` method returns the wrong number of points, because many points are on the edges of some elementary intervals.

**Comment by regislebrun** [comment]
Good idea. We should think a little bit about an architecture allowing to define such intervals without making a test in each method because the penalty involved may be significant.
An option may be to shift a little bit the bounds components to get the *exect* same result using a closed interval, as the floating point numbers are discrete by nature. The interval $[a,b)$ is different from $[a,b-\epsilon]$ for any $\epsilon>0$ in maths, but not for floating points.

**Comment by mbaudin47** [comment]
This issue is linked to #2686. Because it prevents from directly using [LowDiscrepancySequenceImplementation.ComputeLocalDiscrepancy(sample, interval)](https://github.com/openturns/openturns/blob/a7e812e65b5b8150cc36468da9d52a020e8a084e/lib/src/Base/Stat/LowDiscrepancySequenceImplementation.cxx#L152).
